src/hevm/src/EVM/ABI.hs

```diff
@@ -540,6 +542,21 @@ listP parser = between (char '[') (char ']') ((do skipSpaces
                                                   skipSpaces
                                                   return a) `sepBy` (char ','))
 
+data AbiVals = NoVals | CAbi [AbiValue] | SAbi [SWord 256]
+
+decodeBuffer :: [AbiType] -> Buffer -> AbiVals
```

Comment:
This is actually not needed in the end, but I'm inclined to keep it because I think it's pretty useful. Do you agree @MrChico? I'm happy to remove it if you prefer to keep things lean...

Reply:
we can keep it around for a bit